### ISSUE TITLE
How to prepare the data for projects/mbt?

### ISSUE THREAD

**Issue by wentaozhu** [google-research/scenic#254]
The dataset processing is unclear. The readme only shows "Additionally, pre-process the training dataset in the same way as done by the ViViT project [here](https://github.com/google-research/scenic/tree/main/scenic/projects/vivit/data/data.md)." And vivit refers the pre-processing to hmdb pro-processing in dmvr.

From this information, it is pretty hard for users to run the code. The only data processing provided is for hmdb. But MBT uses RGB+spectrogram. Could you please provide a bit of information on how to process the data so that users can run the code?

Thank you so much!

**Comment by wentaozhu** [comment]
Thank you so much, Mostafa! @MostafaDehghani

**Comment by VinKK1998** [comment]
Could you provide a script for generating correspoding tfrecords file from audioset dataset?

**Comment by a-nagrani** [comment]
Hi, sorry but we can't release our data processing scripts. However you can follow the instructions here: https://github.com/deepmind/dmvr/tree/master/examples to create tfrecords files in the correct DMVR format.

**Comment by jayathungek** [comment]
Hi all, I have a few questions about reproducing this project as well:

1. I suppose this means that we have to download the Youtube videos ourselves and apply the pre-processing as per https://github.com/deepmind/dmvr/tree/master/examples - Is that correct?

2. Also, to expand on @wentaozhu's point, MBT supports RGB as well as spectrograms, but in your `projects/mbt/configs/audioset/balanced_audioset_base.py` config file, `config.dataset_configs.tables` only seems to contain spectrogram tfrecords, i.e. `balanced_train.se.melspec.tfrecord.sst@1024`. How can the RGB component of the data be  integrated into this config file?

3. It is also not very clear to me how the dataset split is generated. For training and validation, I assume we use the .csv files provided by [Audioset](https://research.google.com/audioset/download.html), but they make no mention of a test set. Do we just use the same records as the validation set for the test as well?

4. Finally, a minor query about the naming convention of the tfrecord files: what is the significance of the `.sst@1024` at the end of  each record? This is from the config file I mentioned earlier. Does this have something to do with the number of shards the dataset is split into?

Sorry for the barrage of questions, this is my first foray into deep learning research and I'm trying to get an understanding of the best practices etc!

Thank you!

**Comment by uck16m1997** [comment]
Hello everyone, 

I also got some questions for the data preparing process. I couldn't find answers for these in the paper.

1) To get the log mel spectrograms from the audio data. In the process of converting from amplitude to db what reference point was used ?(1, max, median...)

2) I didn't see it get mentioned in the paper but I've seen in the code there is optional zero centering for both the rgb and the spectrogram.  I wanted to know if this was used for the data used to train model checkpoints. 

3) Lastly if it’s possible, where can we get the build configurations that the checkpoints expect of AVTFRecordDatasetFactory ? There seems to be some mismatch that created some confusion. For example when we wan’t to create the dataset using AVTF default num_spec_frame is 5 but checkpoint seems to expect and the paper mentions 8 seconds sampled. I might have seen additional mismatch as well so I would like to be sure.

Sorry for piling on more questions :) I am warming up to these topics so, if you want to point me to additional resources that would be great as well.

Thanks a lot!

**Comment by yangjiangeyjg** [comment]
> Hello everyone,
> 
> I also got some questions for the data preparing process. I couldn't find answers for these in the paper.
> 
> 1. To get the log mel spectrograms from the audio data. In the process of converting from amplitude to db what reference point was used ?(1, max, median...)
> 2. I didn't see it get mentioned in the paper but I've seen in the code there is optional zero centering for both the rgb and the spectrogram.  I wanted to know if this was used for the data used to train model checkpoints.
> 3. Lastly if it’s possible, where can we get the build configurations that the checkpoints expect of AVTFRecordDatasetFactory ? There seems to be some mismatch that created some confusion. For example when we wan’t to create the dataset using AVTF default num_spec_frame is 5 but checkpoint seems to expect and the paper mentions 8 seconds sampled. I might have seen additional mismatch as well so I would like to be sure.
> 
> Sorry for piling on more questions :) I am warming up to these topics so, if you want to point me to additional resources that would be great as well.
> 
> Thanks a lot!

Hi, have you resolved these issues?

**Comment by yangjiangeyjg** [comment]
> Hi all, I have a few questions about reproducing this project as well:
> 
> 1. I suppose this means that we have to download the Youtube videos ourselves and apply the pre-processing as per https://github.com/deepmind/dmvr/tree/master/examples - Is that correct?
> 2. Also, to expand on @wentaozhu's point, MBT supports RGB as well as spectrograms, but in your `projects/mbt/configs/audioset/balanced_audioset_base.py` config file, `config.dataset_configs.tables` only seems to contain spectrogram tfrecords, i.e. `balanced_train.se.melspec.tfrecord.sst@1024`. How can the RGB component of the data be  integrated into this config file?
> 3. It is also not very clear to me how the dataset split is generated. For training and validation, I assume we use the .csv files provided by [Audioset](https://research.google.com/audioset/download.html), but they make no mention of a test set. Do we just use the same records as the validation set for the test as well?
> 4. Finally, a minor query about the naming convention of the tfrecord files: what is the significance of the `.sst@1024` at the end of  each record? This is from the config file I mentioned earlier. Does this have something to do with the number of shards the dataset is split into?
> 
> Sorry for the barrage of questions, this is my first foray into deep learning research and I'm trying to get an understanding of the best practices etc!
> 
> Thank you!

Hi, have you resolved these issues?

**Comment by yangjiangeyjg** [comment]
> Could you provide a script for generating correspoding tfrecords file from audioset dataset?

Hi, have you resolved these issues?

**Comment by yangjiangeyjg** [comment]
> Hi, have you resolved these issues?

Hi, have you resolved these issues?

**Comment by yangjiangeyjg** [comment]
> Hi, sorry but we can't release our data processing scripts. However you can follow the instructions here: https://github.com/deepmind/dmvr/tree/master/examples to create tfrecords files in the correct DMVR format.

Hi, could you disclose the processed data?

**Comment by BDHU** [comment]
@wentaozhu Following this up, any updates on this?

**Comment by a-nagrani** [comment]
No sorry, we cannot release the processed data either.
